Rework common conditions and consumer count

23doors opened the pull request:
Consumers connected should be matched per queue.
While nodes_running and open_connections are common for all queues and it's redundant to check them more then once in a loop.

mylk commented:
Hello @23doors!

Thank you for the contribution to the project!
You are right about "nodes_running", "node_memory_used" and "open_connections".

Regarding the change on "consumers_connected", I think its going to be confusing for existing users as currently its the minimum total connected consumers. On 3ab2602 its changed to minimum connected consumers on each queue.
That means existing users after updating will start receiving alerts about this, due to the context switching of the condition.

As its a nice idea for having that condition on queue level too, I would recommend creating it as a new one. Probably something like "queue_consumers_connected"? Feel free to propose any other name.

Last but not least, I would be great if you could also take a look at the tests,
as they are currently failing. In case you don't have enough time, I can take care of the tests.

Thanks.

23doors commented:
@mylk Fixed tests and added `queue_consumers_connected`. Reverted `consumers_connected`. Also made it so that all conditions are optional (previously it raised exception on first missing config key). 

Should be now mostly backwards compatible - only difference being that it now requires `consumers_connected` etc to be a part of `[Conditions]` instead of `[Conditions:queue_name]`. This is due to these checks being global rather than per queue.

mylk commented:
Hello @23doors 

Thank you for the updates.

Regarding the changes in Dockerfile, I would like to note that this file is also for our repository in dockerhub, which means that there is no "/app" and the application needs to be downloaded.
Can you revert that commit so I can proceed with merging?

Thanks

23doors commented:
@mylk Sorry, I committed that by mistake. I am using this Dockerfile as a build hook in docker hub. This does not require any "/app" folder though. It copies all github source from current branch to /app and installs it from there. As such - this way it doesn't depend on PyPI deployment and is much more conveniet to use. Not to mention it builds automatically even for free docker hub account. But this is not the point of this PR - removed these commits. Still, I would suggest looking into that ;)

mylk commented:
Thanks @23doors, I'll take a look for sure!
